target/linux/ath79/dts/ar7161_fortinet_fap-220-b.dts

```diff
@@ -0,0 +1,183 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "ar7100.dtsi"
+#include "arxxxx_fortinet_loader.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "fortinet,fap-220-b", "qca,ar7161";
+	model = "Fortinet FAP-220-B";
+
+	chosen {
+		bootargs = "console=ttyS0,9600";
+	};
+
+	aliases {
+		led-boot = &led_power;
+		led-failsafe = &led_power;
+		led-running = &led_power;
+		led-upgrade = &led_power;
+		label-mac-device = &eth1;
+	};
+
+	keys {
+		compatible = "gpio-keys";
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+			debounce-interval = <60>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_power: power-green {
+			label = "green:power";
+			gpios = <&gpio 4 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+
+		status-green {
+			label = "green:status";
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+		};
+
+		status-yellow {
+			label = "yellow:status";
+			gpios = <&gpio 0 GPIO_ACTIVE_LOW>;
+			panic-indicator;
+		};
+
+		mode-green {
+			label = "green:mode";
+			gpios = <&gpio 5 GPIO_ACTIVE_LOW>;
+		};
+
+		mode-yellow {
+			label = "yellow:mode";
+			gpios = <&gpio 6 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan2g-green {
+			label = "green:wlan2g";
+			gpios = <&ath9k0 5 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		wlan2g-yellow {
+			label = "yellow:wlan2g";
+			gpios = <&ath9k0 3 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0assoc";
+		};
+
+		wlan5g-green {
+			label = "green:wlan5g";
+			gpios = <&ath9k1 5 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+
+		wlan5g-yellow {
+			label = "yellow:wlan5g";
+			gpios = <&ath9k1 3 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1assoc";
+		};
+	};
+
+	virtual_flash {
+		devices = <&fwconcat0 &fwconcat1 &fwconcat2>;
+	};
+};
+
+&pcie0 {
+	status = "okay";
+
+	ath9k0: wifi@0,11 { /* 2.4 GHz */
+		compatible = "pci168c,0029";
+		reg = <0x8800 0 0 0 0>;
+		ieee80211-freq-limit = <2402000 2482000>;
+		nvmem-cells = <&macaddr_art_120c>, <&cal_art_1000>;
+		nvmem-cell-names = "mac-address", "calibration";
+		mac-address-increment = <1>;
```

Comment:
Would prefer to see the upstream (backported to Openwrt) nvmem `fixed-layout`, with `nvmem-cell-cells` instead of `mac-address-increment`. Example here: https://github.com/openwrt/openwrt/commit/daaa0c1b250498acbf91697dfc75c9d3fbac3596 bindings: https://elixir.bootlin.com/linux/v6.6-rc2/source/Documentation/devicetree/bindings/nvmem/layouts/fixed-layout.yaml, https://elixir.bootlin.com/linux/v6.6-rc2/source/Documentation/devicetree/bindings/nvmem/layouts/fixed-cell.yaml

Reply:
Good idea, although I see some potential for improvement - `mac-base` lacks support for parsing MACs which are written in ASCII but without separators as of now.